app/controllers/medications_controller.rb

```diff
@@ -0,0 +1,27 @@
+class MedicationsController < ApplicationController
+  def new; end
+
+  def search
+    results = SEARCH_RESULTS.med_search_results(params[:medication_name])
+    return @med_hash = results unless results.nil?
+    flash[:warning] = "Sorry, your search did not return any results. Please try another search."
+    redirect_to '/medications/new'
+  end
+
+  def create
+    medication = current_user.medications.create(med_params)
```

Comment:
this will create duplicate records of the same medication if another user has already added it to their own meds list. I think this is where the `find_or_create` piece would come in

Reply:
Ah, good catch. I did open an issue to address the fact that we need code to prevent duplication of medications for the same user, but i did not think about the fact that it would also be adding it to the same Medications table in general.